### PR TITLE
fix masking formula bug and only calculate loss on masked tokens

### PR DESCRIPTION
**Pull request by EmaadKhwaja**
cosine schedule calculates the number of tokens which are UNMASKED

### CODE REVIEW

File: transformer.py [hunk 1]
```diff
@@ -68,15 +68,15 @@ def forward(self, x):
...
+        a_indices = ~mask * z_indices + (mask) * masked_indices
 
         a_indices = torch.cat((sos_tokens, a_indices), dim=1)
 
         target = torch.cat((sos_tokens, z_indices), dim=1)
 
         logits = self.transformer(a_indices)
 
-        return logits, target
+        return logits[~mask], target[~mask]
```
Comment (CUN-bjy) [inline transformer.py:79]: This should be like this?
```suggestion
        return logits[:, ~mask], target[:, ~mask]

```
